File: src/chipschallenge/blockreactions/HintTo.java
package chipschallenge.blockreactions;

import chipschallenge.Block;

public class HintTo extends NoSlipReaction {

    private HintTo() {
    }
    private static HintTo mInstance = null;

    public static synchronized HintTo getInstance() {
        if (mInstance == null) {
            mInstance = new HintTo();
        }
        return mInstance;
    }

    public void react(Block moving, Block standing) {
        game().showHint();
    }

    public boolean canMove(Block moving, Block standing) {
        return true;
    }
}

File: ChipsChallenge.Shared/Blockreactions/HintTo.cs
namespace ChipsChallenge.Shared.Blockreactions
{
    public class HintTo : NoSlipReaction
    {
        private static HintTo mInstance;

        private HintTo()
        {
        }

        public static HintTo Instance
        {
            get
            {
                lock (typeof(HintTo))
                {
                    return mInstance ?? (mInstance = new HintTo());
                }
            }
        }

        public override void React(Block moving, Block standing)
        {
            Game().IsChipOnHintField = true;
        }

        public override bool canMove(Block moving, Block standing)
        {
            return true;
        }
    }
}